### ISSUE TITLE
Create a todo within a normal message

### ISSUE THREAD

**Issue by biva** [mattermost/mattermost-plugin-todo#132]
I see 2 possibilities to create a todo currently:
- A. I write `/todo send @bob Watch this` → in this case, the result is that my original message disappears, only the todo is added to Bob's list and Bob sees a message from **Todo Bot**.
- B. I write `Watch this` in a channel, click on the 3 dots, choose **Add Todo**, choose the user, ... → this is good but a lot of steps.

I'd like to mix A and B:
1. In the channel "News", we have Bob, Alice and me
2. I write `/todo send @bob Watch this` 
3. The todo is added in Bob's list
4. My original message `/todo send @bob Watch this` is still visible in the channel "News": Alice and me can see it.

This would be really useful to give some context and allow follow-up. For example, I could write:
`Hey, I have watched the latest report from CrazyNews, and it's interesting`
And reply to this message writing:  `/todo send @bob Watch it`

**Comment by larkox** [comment]
If you were to create a todo like that, Bob will just receive the "Wath it" part of the todo. In order for a todo to be "linked" to a message, you have to add the permalink. To do that, you can get the message you want to reference (for example, the news link), create a todo, and remove the content (just keeping the permalin) and just add "Watch this". Would that solve the problem?

Do you see any other way to ease your issue?

CC. @aaronrothschild

**Comment by biva** [comment]
Thanks @larkox for your feedback. I understand your point with the permalink, which is indeed important. In this case, the todo should just include `Watch it` and the permalink to the discussion: Bob can click on the link and he will land on the discussion to see what it refers to.

My objective is to make it very easy and straight forward to create a todo, without any additional step than writing a message. I understand that your solution involves more steps. Am I wrong?

**Comment by larkox** [comment]
Yes, my solution involves more steps. But I struggle to see how to make clear the options to the user. For example, how you differentiate between:
- `/todo send @bob Do the dishes` (a todo with reference to nothing)
- `/todo send @bob Watch this video` (a todo with reference to a message)

**Comment by biva** [comment]
I wouldn't make any difference: I would include the permalink in both cases. For the dishes, Bob won't care about the permalink. And for the video, he will use it.

**Comment by biva** [comment]
Hello, I see that some people are working on this request, thank you so much! Could you please update us on the status? Thank you again!
